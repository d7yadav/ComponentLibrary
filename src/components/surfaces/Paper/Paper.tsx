import React, { forwardRef, memo, useCallback, useState } from 'react';

import {
  PAPER_VARIANTS,
  PAPER_CORNERS,
  PAPER_SURFACES,
  PAPER_SIZES,
  PAPER_GRADIENTS,
  ANIMATION_DURATIONS,
  ACCESSIBILITY_CONSTANTS,
} from './Paper.constants';
import { StyledPaper, paperHoverKeyframes } from './Paper.styles';
import type { PaperProps } from './Paper.types';

/**
 * Enhanced Paper component with multiple variants, surfaces, and accessibility features
 * 
 * Features:
 * - 5 variants: elevation, outlined, filled, glass, gradient
 * - 24 elevation levels (0-24) with proper shadows
 * - 5 corner styles: none, small, medium, large, circular
 * - 3 surface treatments: flat, concave, convex
 * - Glass morphism effects with backdrop blur
 * - Gradient overlays with theme integration
 * - Interactive hover states with elevation changes
 * - Responsive padding and sizing
 * - Print-friendly styling
 * - WCAG 2.1 AA accessibility compliance
 * - Dark theme optimizations with OLED support
 */
const PaperComponent = forwardRef<HTMLDivElement, PaperProps>(({
  variant = PAPER_VARIANTS.elevation,
  elevation = 1,
  corners = PAPER_CORNERS.medium,
  surface = PAPER_SURFACES.flat,
  size = PAPER_SIZES.comfortable,
  gradient = PAPER_GRADIENTS.primary,
  interactive = false,
  glassMorphism = false,
  responsive = false,
  printFriendly = false,
  hoverElevation,
  pressedElevation,
  backgroundColor,
  borderColor,
  borderWidth,
  padding,
  margin,
  animationDuration = ANIMATION_DURATIONS.normal,
  transitions = true,
  overflow,
  maxWidth,
  maxHeight,
  minWidth,
  minHeight,
  zIndex,
  children,
  className,
  sx,
  onClick,
  onMouseEnter,
  onMouseLeave,
  onFocus,
  onBlur,
  'aria-label': ariaLabel,
  'aria-describedby': ariaDescribedBy,
  'aria-labelledby': ariaLabelledBy,
  role,
  tabIndex,
  id,
  'data-testid': dataTestId = 'paper',
  ...other
}, ref) => {
  const [isHovered, setIsHovered] = useState(false);
  const [, setIsFocused] = useState(false);

  // Handle mouse enter
  const handleMouseEnter = useCallback((event: React.MouseEvent<HTMLDivElement>) => {
    if (interactive) {
      setIsHovered(true);
    }
    onMouseEnter?.(event);
  }, [interactive, onMouseEnter]);

  // Handle mouse leave
  const handleMouseLeave = useCallback((event: React.MouseEvent<HTMLDivElement>) => {
    if (interactive) {
      setIsHovered(false);
    }
    onMouseLeave?.(event);
  }, [interactive, onMouseLeave]);

  // Handle focus
  const handleFocus = useCallback((event: React.FocusEvent<HTMLDivElement>) => {
    if (interactive) {
      setIsFocused(true);
    }
    onFocus?.(event);
  }, [interactive, onFocus]);

  // Handle blur
  const handleBlur = useCallback((event: React.FocusEvent<HTMLDivElement>) => {
    if (interactive) {
      setIsFocused(false);
    }
    onBlur?.(event);
  }, [interactive, onBlur]);

  // Handle click
  const handleClick = useCallback((event: React.MouseEvent<HTMLDivElement>) => {
    if (interactive && onClick) {
      onClick(event);
    }
  }, [interactive, onClick]);

  // Handle keyboard events for interactive papers
  const handleKeyDown = useCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
    if (interactive && (event.key === 'Enter' || event.key === ' ')) {
      event.preventDefault();
      handleClick(event as any);
    }
  }, [interactive, handleClick]);

  // Determine accessibility props
  const getAccessibilityProps = () => {
    const baseProps = {
      'aria-label': ariaLabel,
      'aria-describedby': ariaDescribedBy,
      'aria-labelledby': ariaLabelledBy,
      role: role || (interactive ? ACCESSIBILITY_CONSTANTS.interactiveRole : undefined),
      ...(id && { id }),
      'data-testid': dataTestId,
    };

    if (interactive) {
      return {
        ...baseProps,
        tabIndex: tabIndex !== undefined ? tabIndex : 0,
        onKeyDown: handleKeyDown,
        'aria-pressed': onClick ? false : undefined,
      };
    }

    return {
      ...baseProps,
      ...(tabIndex !== undefined && { tabIndex }),
    };
  };

  // Determine hover and pressed elevations
  const getHoverElevation = () => {
    if (hoverElevation !== undefined) return hoverElevation;
    if (!interactive) return elevation;
    return Math.min(elevation + 2, 24) as typeof elevation;
  };

  const getPressedElevation = () => {
    if (pressedElevation !== undefined) return pressedElevation;
    if (!interactive) return elevation;
    return Math.max(elevation - 1, 0) as typeof elevation;
  };

  // Component props for styled component
  const styledProps = {
    customVariant: variant,
    customElevation: elevation,
    corners,
    surface,
    size,
    gradient,
    interactive,
    glassMorphism: variant === 'glass' ? glassMorphism : false,
    responsive,
    printFriendly,
    hoverElevation: getHoverElevation(),
    pressedElevation: getPressedElevation(),
    backgroundColor,
    borderColor,
    borderWidth,
    padding,
    margin,
    animationDuration,
    transitions,
    overflow,
    maxWidth,
    maxHeight,
    minWidth,
    minHeight,
    zIndex,
  };

  // Additional classes for special effects
  const getAdditionalClasses = () => {
    const classes = [];
    
    if (variant === 'glass' && glassMorphism && isHovered) {
      classes.push('paper-glass-shimmer');
    }
    
    if (className) {
      classes.push(className);
    }
    
    return classes.join(' ') || undefined;
  };

  return (
    <>
      <style>{paperHoverKeyframes}</style>
      <StyledPaper
        ref={ref}
        className={getAdditionalClasses()}
        sx={sx}
        onClick={handleClick}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        onFocus={handleFocus}
        onBlur={handleBlur}
        {...getAccessibilityProps()}
        {...styledProps}
        {...other}
      >
        {children}
      </StyledPaper>
    </>
  );
});

PaperComponent.displayName = 'Paper';

// Export memoized component for performance optimization
export const Paper = memo(PaperComponent);