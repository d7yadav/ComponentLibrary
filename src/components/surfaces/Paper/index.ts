export { Paper } from './Paper';
export type {
  PaperProps,
  PaperVariant,
  PaperElevation,
  PaperCorners,
  PaperSurface,
  PaperSize,
  PaperGradient,
  PaperStyleProps,
  PaperSizeConfig,
  PaperCornerConfig,
  PaperSurfaceConfig,
  PaperGradientConfig,
  ResponsivePaperConfig,
} from './Paper.types';

export {
  PAPER_VARIANTS,
  PAPER_CORNERS,
  PAPER_SURFACES,
  PAPER_SIZES,
  PAPER_GRADIENTS,
  PAPER_SIZE_CONFIGS,
  PAPER_CORNER_CONFIGS,
  PAPER_SURFACE_CONFIGS,
  PAPER_GRADIENT_CONFIGS,
  PAPER_GRADIENT_CONFIGS_DARK,
  ELEVATION_SHADOWS,
  ANIMATION_DURATIONS,
  ANIMATION_EASINGS,
  GLASS_MORPHISM_CONFIG,
  GLASS_MORPHISM_CONFIG_DARK,
  PRINT_STYLES,
  RESPONSIVE_BREAKPOINTS,
  ACCESSIBILITY_CONSTANTS,
  Z_INDEX_LEVELS,
} from './Paper.constants';

export {
  StyledPaper,
  paperHoverKeyframes,
  paperCssVars,
  getElevationShadow,
  getCornerRadius,
  shouldReduceMotion,
} from './Paper.styles';