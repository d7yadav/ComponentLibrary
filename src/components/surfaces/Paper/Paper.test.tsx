/**
 * @fileoverview Paper Component Tests
 * @author AI Generated Tests - 2025-07-13
 */

import { ThemeProvider , createTheme } from '@mui/material/styles';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe, toHaveNoViolations } from 'jest-axe';

import { Paper } from './Paper';
import type { PaperProps } from './Paper.types';

// Add jest-axe matcher
expect.extend(toHaveNoViolations);

// Test theme
const testTheme = createTheme();

// Test wrapper component
const TestWrapper = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={testTheme}>{children}</ThemeProvider>
);

// Helper function to render component with theme
const renderWithTheme = (ui: React.ReactElement, options = {}) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

describe('Paper Component', () => {
  // Basic rendering tests
  describe('Rendering', () => {
    it('renders without crashing', () => {
      renderWithTheme(<Paper>Test content</Paper>);
      expect(screen.getByText('Test content')).toBeInTheDocument();
    });

    it('renders children correctly', () => {
      const testContent = 'Test Paper Content';
      renderWithTheme(<Paper>{testContent}</Paper>);
      expect(screen.getByText(testContent)).toBeInTheDocument();
    });

    it('renders with custom test id', () => {
      const testId = 'custom-paper';
      renderWithTheme(
        <Paper data-testid={testId}>Test</Paper>
      );
      expect(screen.getByTestId(testId)).toBeInTheDocument();
    });

    it('applies custom className', () => {
      const customClass = 'custom-class';
      renderWithTheme(
        <Paper className={customClass} data-testid="paper">Test</Paper>
      );
      expect(screen.getByTestId('paper')).toHaveClass(customClass);
    });
  });

  // Props testing
  describe('Props', () => {
    describe('variant prop', () => {
      const variants: PaperProps['variant'][] = [
        'elevation', 'outlined', 'filled', 'glass', 'gradient'
      ];

      variants.forEach((variant) => {
        it(`renders ${variant} variant correctly`, () => {
          renderWithTheme(
            <Paper variant={variant} data-testid="variant-test">
              {variant} variant
            </Paper>
          );
          const element = screen.getByTestId('variant-test');
          expect(element).toBeInTheDocument();
        });
      });
    });

    describe('elevation prop', () => {
      const elevations: PaperProps['elevation'][] = [
        0, 1, 2, 3, 4, 6, 8, 12, 16, 24
      ];

      elevations.forEach((elevation) => {
        it(`renders elevation ${elevation} correctly`, () => {
          renderWithTheme(
            <Paper elevation={elevation} data-testid="elevation-test">
              Elevation {elevation}
            </Paper>
          );
          const element = screen.getByTestId('elevation-test');
          expect(element).toBeInTheDocument();
        });
      });
    });

    describe('corners prop', () => {
      const corners: PaperProps['corners'][] = [
        'none', 'small', 'medium', 'large', 'circular'
      ];

      corners.forEach((corner) => {
        it(`renders ${corner} corners correctly`, () => {
          renderWithTheme(
            <Paper corners={corner} data-testid="corners-test">
              {corner} corners
            </Paper>
          );
          const element = screen.getByTestId('corners-test');
          expect(element).toBeInTheDocument();
        });
      });
    });

    describe('surface prop', () => {
      const surfaces: PaperProps['surface'][] = [
        'flat', 'concave', 'convex'
      ];

      surfaces.forEach((surface) => {
        it(`renders ${surface} surface correctly`, () => {
          renderWithTheme(
            <Paper surface={surface} data-testid="surface-test">
              {surface} surface
            </Paper>
          );
          const element = screen.getByTestId('surface-test');
          expect(element).toBeInTheDocument();
        });
      });
    });

    it('handles square prop correctly', () => {
      renderWithTheme(
        <Paper square data-testid="square-test">
          Square Paper
        </Paper>
      );
      const element = screen.getByTestId('square-test');
      expect(element).toBeInTheDocument();
    });

    it('handles interactive prop correctly', () => {
      const handleClick = vi.fn();
      renderWithTheme(
        <Paper interactive onClick={handleClick} data-testid="interactive-test">
          Interactive Paper
        </Paper>
      );
      const element = screen.getByTestId('interactive-test');
      fireEvent.click(element);
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('handles disabled state correctly', () => {
      const handleClick = vi.fn();
      renderWithTheme(
        <Paper disabled onClick={handleClick} data-testid="disabled-test">
          Disabled Paper
        </Paper>
      );
      
      const element = screen.getByTestId('disabled-test');
      fireEvent.click(element);
      expect(handleClick).not.toHaveBeenCalled();
    });

    it('handles gradient prop correctly', () => {
      renderWithTheme(
        <Paper gradient="primary" data-testid="gradient-test">
          Gradient Paper
        </Paper>
      );
      const element = screen.getByTestId('gradient-test');
      expect(element).toBeInTheDocument();
    });

    it('handles size prop correctly', () => {
      renderWithTheme(
        <Paper size="large" data-testid="size-test">
          Large Paper
        </Paper>
      );
      const element = screen.getByTestId('size-test');
      expect(element).toBeInTheDocument();
    });
  });

  // Interaction testing
  describe('Interactions', () => {
    it('handles click events when interactive', async () => {
      const user = userEvent.setup();
      const handleClick = vi.fn();
      
      renderWithTheme(
        <Paper interactive onClick={handleClick} data-testid="clickable">
          Clickable Paper
        </Paper>
      );
      
      const element = screen.getByTestId('clickable');
      await user.click(element);
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('handles keyboard navigation when interactive', async () => {
      const user = userEvent.setup();
      const handleClick = vi.fn();
      
      renderWithTheme(
        <Paper interactive onClick={handleClick} tabIndex={0} data-testid="keyboard-nav">
          Keyboard navigable Paper
        </Paper>
      );
      
      const element = screen.getByTestId('keyboard-nav');
      element.focus();
      expect(element).toHaveFocus();
      
      await user.keyboard('{Enter}');
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('handles hover states when interactive', () => {
      renderWithTheme(
        <Paper interactive data-testid="hoverable">
          Hoverable Paper
        </Paper>
      );
      
      const element = screen.getByTestId('hoverable');
      fireEvent.mouseEnter(element);
      expect(element).toBeInTheDocument();
      
      fireEvent.mouseLeave(element);
      expect(element).toBeInTheDocument();
    });

    it('does not respond to interactions when disabled', async () => {
      const user = userEvent.setup();
      const handleClick = vi.fn();
      
      renderWithTheme(
        <Paper disabled onClick={handleClick} data-testid="disabled-interaction">
          Disabled Paper
        </Paper>
      );
      
      const element = screen.getByTestId('disabled-interaction');
      await user.click(element);
      expect(handleClick).not.toHaveBeenCalled();
    });
  });

  // Accessibility testing
  describe('Accessibility', () => {
    it('meets WCAG accessibility guidelines', async () => {
      const { container } = renderWithTheme(
        <Paper>Accessible Paper</Paper>
      );
      
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });

    it('provides proper ARIA attributes when interactive', () => {
      renderWithTheme(
        <Paper 
          interactive
          role="button"
          aria-label="Interactive paper"
          aria-describedby="description"
          data-testid="aria-paper"
        >
          ARIA Paper
        </Paper>
      );
      
      const element = screen.getByTestId('aria-paper');
      expect(element).toHaveAttribute('role', 'button');
      expect(element).toHaveAttribute('aria-label', 'Interactive paper');
      expect(element).toHaveAttribute('aria-describedby', 'description');
    });

    it('supports keyboard navigation', () => {
      renderWithTheme(
        <Paper interactive tabIndex={0} data-testid="keyboard-paper">
          Keyboard accessible Paper
        </Paper>
      );
      
      const element = screen.getByTestId('keyboard-paper');
      element.focus();
      expect(element).toHaveFocus();
    });

    it('indicates disabled state with aria-disabled', () => {
      renderWithTheme(
        <Paper disabled data-testid="disabled-aria">
          Disabled Paper
        </Paper>
      );
      
      const element = screen.getByTestId('disabled-aria');
      expect(element).toHaveAttribute('aria-disabled', 'true');
    });
  });

  // Theme integration testing
  describe('Theme Integration', () => {
    it('respects theme customizations', () => {
      const customTheme = createTheme({
        components: {
          MuiPaper: {
            styleOverrides: {
              root: {
                backgroundColor: 'red',
              },
            },
          },
        },
      });

      render(
        <ThemeProvider theme={customTheme}>
          <Paper data-testid="themed-paper">
            Themed Paper
          </Paper>
        </ThemeProvider>
      );

      expect(screen.getByTestId('themed-paper')).toBeInTheDocument();
    });

    it('supports dark mode', () => {
      const darkTheme = createTheme({
        palette: {
          mode: 'dark',
        },
      });

      render(
        <ThemeProvider theme={darkTheme}>
          <Paper data-testid="dark-paper">
            Dark Mode Paper
          </Paper>
        </ThemeProvider>
      );

      expect(screen.getByTestId('dark-paper')).toBeInTheDocument();
    });

    it('applies elevation shadows correctly', () => {
      renderWithTheme(
        <Paper elevation={8} data-testid="shadow-paper">
          Elevated Paper
        </Paper>
      );

      expect(screen.getByTestId('shadow-paper')).toBeInTheDocument();
    });
  });

  // Performance testing
  describe('Performance', () => {
    it('renders efficiently with many instances', () => {
      const manyPapers = Array.from({ length: 50 }, (_, i) => (
        <Paper key={i} elevation={i % 5}>
          Paper {i}
        </Paper>
      ));

      const { container } = renderWithTheme(<div>{manyPapers}</div>);
      expect(container).toBeInTheDocument();
    });

    it('memoizes properly to prevent unnecessary re-renders', () => {
      const { rerender } = renderWithTheme(
        <Paper elevation={2} data-testid="memo-test">
          Memoized Paper
        </Paper>
      );

      const element = screen.getByTestId('memo-test');
      expect(element).toBeInTheDocument();

      // Re-render with same props should not cause issues
      rerender(
        <TestWrapper>
          <Paper elevation={2} data-testid="memo-test">
            Memoized Paper
          </Paper>
        </TestWrapper>
      );

      expect(screen.getByTestId('memo-test')).toBeInTheDocument();
    });

    it('handles elevation changes efficiently', () => {
      const { rerender } = renderWithTheme(
        <Paper elevation={1} data-testid="elevation-change">
          Changing Elevation
        </Paper>
      );

      let element = screen.getByTestId('elevation-change');
      expect(element).toBeInTheDocument();

      rerender(
        <TestWrapper>
          <Paper elevation={24} data-testid="elevation-change">
            Changing Elevation
          </Paper>
        </TestWrapper>
      );

      element = screen.getByTestId('elevation-change');
      expect(element).toBeInTheDocument();
    });
  });

  // Edge cases and error handling
  describe('Edge Cases', () => {
    it('handles empty children gracefully', () => {
      renderWithTheme(<Paper data-testid="empty-paper" />);
      expect(screen.getByTestId('empty-paper')).toBeInTheDocument();
    });

    it('handles null children gracefully', () => {
      renderWithTheme(
        <Paper data-testid="null-paper">{null}</Paper>
      );
      expect(screen.getByTestId('null-paper')).toBeInTheDocument();
    });

    it('handles undefined props gracefully', () => {
      renderWithTheme(
        <Paper 
          variant={undefined as any}
          elevation={undefined as any}
          data-testid="undefined-props"
        >
          Undefined Props
        </Paper>
      );
      expect(screen.getByTestId('undefined-props')).toBeInTheDocument();
    });

    it('handles extreme elevation values', () => {
      renderWithTheme(
        <Paper elevation={999 as any} data-testid="extreme-elevation">
          Extreme Elevation
        </Paper>
      );
      expect(screen.getByTestId('extreme-elevation')).toBeInTheDocument();
    });

    it('handles complex nested content', () => {
      renderWithTheme(
        <Paper data-testid="nested-content">
          <div>
            <h1>Nested Title</h1>
            <p>Nested paragraph with <strong>bold</strong> text</p>
            <ul>
              <li>List item 1</li>
              <li>List item 2</li>
            </ul>
          </div>
        </Paper>
      );
      expect(screen.getByTestId('nested-content')).toBeInTheDocument();
      expect(screen.getByText('Nested Title')).toBeInTheDocument();
    });
  });

  // Snapshots
  describe('Snapshots', () => {
    it('matches snapshot for default props', () => {
      const { container } = renderWithTheme(
        <Paper>Default Paper</Paper>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for all variants', () => {
      const variants: PaperProps['variant'][] = [
        'elevation', 'outlined', 'filled', 'glass', 'gradient'
      ];

      variants.forEach((variant) => {
        const { container } = renderWithTheme(
          <Paper variant={variant}>{variant} Paper</Paper>
        );
        expect(container.firstChild).toMatchSnapshot(`variant-${variant}`);
      });
    });

    it('matches snapshot for different elevations', () => {
      const elevations = [0, 1, 4, 8, 16, 24];

      elevations.forEach((elevation) => {
        const { container } = renderWithTheme(
          <Paper elevation={elevation}>Elevation {elevation}</Paper>
        );
        expect(container.firstChild).toMatchSnapshot(`elevation-${elevation}`);
      });
    });
  });
});