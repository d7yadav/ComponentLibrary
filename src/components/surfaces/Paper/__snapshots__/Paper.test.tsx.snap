// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Paper Component > Snapshots > matches snapshot for all variants > variant-elevation 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for all variants > variant-filled 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for all variants > variant-glass 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for all variants > variant-gradient 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for all variants > variant-outlined 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for default props 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for different elevations > elevation-0 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for different elevations > elevation-1 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for different elevations > elevation-4 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for different elevations > elevation-8 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for different elevations > elevation-16 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;

exports[`Paper Component > Snapshots > matches snapshot for different elevations > elevation-24 1`] = `
<style>
  
  @keyframes paper-hover-elevation {
    0% {
      transform: translateY(0px);
    }
    100% {
      transform: translateY(-1px);
    }
  }

  @keyframes paper-press-elevation {
    0% {
      transform: translateY(-1px);
    }
    100% {
      transform: translateY(0px);
    }
  }

  @keyframes paper-glass-shimmer {
    0% {
      background-position: -200% 0;
    }
    100% {
      background-position: 200% 0;
    }
  }

  .paper-glass-shimmer::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: linear-gradient(
      90deg,
      transparent 0%,
      rgba(255, 255, 255, 0.4) 50%,
      transparent 100%
    );
    background-size: 200% 100%;
    animation: paper-glass-shimmer 2s infinite;
    border-radius: inherit;
    pointer-events: none;
  }

</style>
`;
